editor/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -89,8 +108,13 @@ class HierarchicalTermSelector extends Component {
 				.then( resolve, ( xhr ) => {
 					const errorCode = xhr.responseJSON && xhr.responseJSON.code;
 					if ( errorCode === 'term_exists' ) {
-						this.addRequest = new Model( { id: xhr.responseJSON.data } ).fetch();
-						return this.addRequest.then( resolve, reject );
+						// search the new category created since last fetch
+						this.addRequest = new Model().fetch(
```

Comment:
Why did you change the request here? Thinking that requesting using the `id` is more reliable, isn't it?

Reply:
The bug was that the server when returns the 409 answer does not return the term. So we were generating an invalid request GET http://src.wordpress-develop.test/wp-json/wp/v2/categories/[object%20Object] 404 (Not Found). Requesting by id would be better but I think we can not do that because we don't know the id yet :(

Reply:
Did you try to inspect the `409` response? I think when I did this, the response body used to contain the conflicting `id`.

Reply:
Yes I tried, and unfortunately, we are out of luck the response just contains "{"code":"term_exists","message":"A term with the name provided already exists with this parent.","data":{"status":409}}". Maybe something changed in the API meanwhile.

Reply:
But this means we may have an issue here, what guarantees that the received category is the right one? I think we should compare the `name` once we retrieve the results instead of just picking the first one. Imagine two categories at the same level called "cat" and "cat 2", I believe if you try to insert "cat", the order is not guaranteed in the API response and we may insert "cat 2" instead right?

Reply:
Yes, you are right, we should add a client-side filter on the search results. I will make the change :)

Reply:
The change was applied, we should be handling all the cases now.